src/OmniSharp.Host/MSBuild/Discovery/Providers/MonoInstanceProvider.cs

```diff
@@ -20,6 +21,32 @@ public override ImmutableArray<MSBuildInstance> GetInstances()
                 return NoInstances;
             }
 
+            // Don't resolve to MSBuild assemblies under the installed Mono unless OmniSharp
+            // is actually running under the installed Mono.
+            var monoRuntimePath = PlatformHelper.GetMonoRuntimePath();
```

Comment:
Is this obvious? Maybe explain a bit further what goes wrong if this is not the case?

Reply:
None of this is obvious. :smile: But you're right, I'll add a little bit more detail about how it can go wrong.